nav/load_waypoints/scripts/navigate_waypoints.py

```diff
@@ -56,8 +59,11 @@ def populate_waypoint_dict(self):
             gps_info = rospy.wait_for_message('gps/fix', NavSatFix)
             # Append the starting gps coordinate to the waypoints dict as the final waypoint
             last_coord_idx = len(self.waypoints) 
-            self.waypoints[last_coord_idx] = {
-                'id': last_coord_idx, 'longitude': gps_info.longitude, 'latitude': gps_info.latitude, 'description': 'Initial start location', 'frame_id': 'odom'}
+ 
+            # If waypoints are reported in longitudes and latitudes, append a final waypoint to return to the start
+            if self.waypoints[0].get("longitude"):
```

Comment:
Wyhy was this change necessary ?

Reply:
if you mean the if statement in line 64, its the wraparound we need